### ISSUE TITLE
Cannot hot reload on an iOS device in airplane mode

### ISSUE THREAD

**Issue by collinjackson** [flutter/flutter#10775]
## Steps to Reproduce

Put iOS device into airplane mode.

```
$ flutter run -d i
Launching lib/main.dart on iPhone 7 Plus in debug mode...
CoreSimulatorBridge: Requesting launch of io.flutter.firebaseAnalyticsExample with options: {
	    arguments =     (
	        "--enable-dart-profiling",
	        "--flx=/Users/jackson/git/plugins/packages/firebase_analytics/example/build/app.flx",
	        "--dart-main=/Users/jackson/git/plugins/packages/firebase_analytics/example/lib/main.dart",
	        "--packages=/Users/jackson/git/plugins/packages/firebase_analytics/example/.packages",
	        "--enable-checked-mode",
	        "--observatory-port=8100",
	        "--diagnostic-port=8101"
	    );
	    environment =     {
	    };
	}
CoreSimulatorBridge: Beginning launch sequence for bundle 'io.flutter.firebaseAnalyticsExample'
		retryTimeout: 120.000000 (default write com.apple.CoreSimulatorBridge LaunchRetryTimeout <value>)
		bootTimeout: 300.000000 (default write com.apple.CoreSimulatorBridge BootRetryTimeout <value>)
		bootLeeway: 120.000000 (default write com.apple.CoreSimulatorBridge BootLeeway <value>)
		Note: Use 'xcrun simctl spawn booted defaults write <domain> <key> <value>' to modify defaults in the booted Simulator device.
	Simulator booted at: 2017-06-16 03:31:39 +0000
	Current time: 2017-06-16 16:49:27 +0000
	Within boot leeway: NO
CoreSimulatorBridge: Launch attempt #0 for bundle 'io.flutter.firebaseAnalyticsExample', have been trying to launch for 0 seconds. Modern launch services = YES
CoreSimulatorBridge: Got LSBundleProxy for 'io.flutter.firebaseAnalyticsExample', sequenceNumber 272, cacheGUID 6246FB11-3540-42BF-9900-452B5BAEC9DF
CoreSimulatorBridge: [Common] [FBSSystemService][0x4a37] Sending request to open "io.flutter.firebaseAnalyticsExample"
Unhandled exception:
Exception: Timeout while attempting to retrieve URL for Observatory
#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:1108)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:41)
#2      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
#3      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:375)
#4      _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:394)
#5      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)
```

## Flutter Doctor

[✓] Flutter (on Mac OS X 10.12.4 16E195, locale en-US, channel master)
    • Flutter at /Users/jackson/git/flutter
    • Framework revision 3db402ec98 (11 hours ago), 2017-06-15 23:01:25 -0700
    • Engine revision f741647d49
    • Tools Dart version 1.24.0-dev.6.7

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/jackson/Library/Android/sdk/
    • Platform android-25, build-tools 25.0.3
    • ANDROID_HOME = /Users/jackson/Library/Android/sdk/
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[-] iOS toolchain - develop for iOS devices (Xcode 8.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.1, Build version 8E1000a
    ✗ libimobiledevice and ideviceinstaller are not installed or require updating. To update, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.1
    • CocoaPods version 1.0.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1.3)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4424.63

[✓] Connected devices
    •               • 14710ae2e8497590643751c5532d907ed1ff2eed • ios • iOS  ()
    • iPhone 7 Plus • F8A917C4-9848-49C2-8F0D-664B49E0B635     • ios • iOS 10.3 (simulator)

@cbracken

**Comment by collinjackson** [comment]
It looks like the solution is in the `flutter doctor`. Testing it now.

**Comment by collinjackson** [comment]
I reinstalled Xcode, homebrew, libimobiledevice, ideviceinstaller, applied this [workaround](https://github.com/phonegap/ios-deploy/issues/292) for "unable to locate DeviceSupport", and also ran `sudo chown -R $USER /var/root/` because otherwise Perl was having trouble locating strict.pm because a root-only directory was in the include path.

After doing all that airplane hot reload works great and `flutter devices` can see the name of my device now (before it couldn't). Yay!

Closing as I don't see an action to be taken here (though the amount of effort it took to upgrade was considerable).
